feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts

```diff
@@ -297,4 +311,122 @@ export class ConfiguratorStorefrontUtilsService {
+   */
+  getVerticallyScrolledPixels(): number | undefined {
+    if (this.windowRef.isBrowser()) {
+      return this.windowRef.nativeWindow?.scrollY;
+    }
+    return undefined;
+  }
+
+  /**
+   * Verifies whether the element has a scrollbar.
+   *
+   * @param {string} querySelector - Element query selector
+   * @returns {boolean} - 'True', if the element has a scrollbar, otherwise 'false'
+   */
+  hasScrollbar(querySelector: string): boolean {
+    const element = this.getElement(querySelector);
+    if (element) {
+      return element.scrollHeight > element.clientHeight;
+    }
+    return false;
+  }
+
+  protected isInViewport(element: HTMLElement | undefined): boolean {
+    if (element) {
+      const bounding = element.getBoundingClientRect();
+      const height = element.offsetHeight;
+      const width = element.offsetWidth;
+
+      return (
+        bounding.top >= -height &&
+        bounding.left >= -width &&
+        bounding.right <=
+          (this.windowRef.nativeWindow?.innerWidth || element.clientWidth) +
+            width &&
+        bounding.bottom <=
+          (this.windowRef.nativeWindow?.innerHeight || element.clientHeight) +
+            height
+      );
+    }
+    return false;
+  }
+
+  protected getHeight(querySelector: string): number {
+    const element = this.getElement(querySelector);
+    const isElementInViewport = this.isInViewport(element);
+    if (isElementInViewport && element?.offsetHeight) {
+      return element?.offsetHeight;
+    }
+    return 0;
+  }
+
+  /**
+   * Retrieves the actual height of the spare viewport.
+   *
+   * SPA header, variant configuration overview header and "Add to cart" button occupy certain height of the viewport, that's why
+   * if SPA header, variant configuration overview header and "Add to cart" button are in the viewport,
+   * they will be subtracted from the actual viewport height.
+   *
+   * @returns {number} - Height of the spare viewport.
+   */
+  getSpareViewportHeight(): number {
+    if (this.windowRef.isBrowser()) {
+      const spaHeaderHeight = this.getHeight('header');
+      const ovHeaderHeight = this.getHeight('.VariantConfigOverviewHeader');
+      const addToCartHeight =
+        this.getHeight('cx-configurator-add-to-cart-button') !== 0
+          ? this.getHeight('cx-configurator-add-to-cart-button')
+          : 82;
+
+      const occupiedHeight =
+        spaHeaderHeight + ovHeaderHeight + addToCartHeight * 2;
+
+      return this.windowRef.nativeWindow
+        ? this.windowRef.nativeWindow.innerHeight - occupiedHeight
+        : 0;
+    }
+    return 0;
+  }
+
+  /**
+   * Syncs scrolling.
+   *
+   * @param {string} querySelector - Element query selector
+   * @param {HTMLElement | undefined} element - Element that should be visible within the scrollable element.
+   */
+  syncScroll(querySelector: string, element: HTMLElement | undefined): void {
```

Comment:
Please also try to improve the test coverage of this class. Statements and branch coverage is down significantly

Reply:
I even talked to Sebastian to find a better name but he did not have a better solution as it is.
Do you have a better name for this method?

Reply:
Done!